code/modules/industrial_lift/tram/tram_lift_master.dm

```diff
@@ -219,3 +228,102 @@
+
+	for(var/obj/effect/landmark/tram/nav/candidate_beacon in GLOB.tram_landmarks[beacon_type])
+		if(candidate_beacon.z != origin.z || candidate_beacon.z != nav_beacon.z)
+			continue
+
+		switch(nav_beacon.dir)
+			if(EAST, WEST)
+				if(candidate_beacon.y != nav_beacon.y)
+					continue
+				else if(candidate_beacon.x < nav_beacon.x)
+					inbound_candidates += candidate_beacon
+				else
+					outbound_candidates += candidate_beacon
+			if(NORTH, SOUTH)
+				if(candidate_beacon.x != nav_beacon.x)
+					continue
+				else if(candidate_beacon.y < nav_beacon.y)
+					inbound_candidates += candidate_beacon
+				else
+					outbound_candidates += candidate_beacon
+
+	switch(travel_dir)
+		if(INBOUND)
+			var/obj/effect/landmark/tram/nav/selected = get_closest_atom(/obj/effect/landmark/tram/nav, inbound_candidates, origin)
+			if(selected)
+				return selected
+			stack_trace("No inbound beacon candidate found for [origin]. Cancelling dispatch.")
+			return FALSE
+
+		if(OUTBOUND)
+			var/obj/effect/landmark/tram/nav/selected = get_closest_atom(/obj/effect/landmark/tram/nav, outbound_candidates, origin)
+			if(selected)
+				return selected
+			stack_trace("No outbound beacon candidate found for [origin]. Cancelling dispatch.")
+			return FALSE
+
+		else
+			stack_trace("Tram receieved invalid travel direction [travel_dir]. Cancelling dispatch.")
+
+	return FALSE
+
+/**
+ * Moves the tram when hit by an immovable rod
+ *
+ * Tells the individual tram parts where to actually go and has an extra safety checks
+ * incase multiple inputs get through, preventing conflicting directions and the tram
+ * literally ripping itself apart. all of the actual movement is handled by SStramprocess
+ *
+ * Arguments: collided_rod (the immovable rod that hit the tram)
+ */
+/datum/lift_master/tram/proc/rod_collision(obj/effect/immovablerod/collided_rod)
+	if(!is_operational)
+		return
+	var/rod_velocity_sign
+	// Determine inbound or outbound
+	if(collided_rod.dir & (NORTH|SOUTH))
+		rod_velocity_sign = collided_rod.dir & NORTH ? OUTBOUND : INBOUND
+	else
+		rod_velocity_sign = collided_rod.dir & EAST ? OUTBOUND : INBOUND
+
+	var/obj/effect/landmark/tram/nav/push_destination = closest_nav_in_travel_dir(origin = nav_beacon, travel_dir = rod_velocity_sign, beacon_type = IMMOVABLE_ROD_DESTINATIONS)
+	if(!push_destination)
+		return
+	travel_direction = get_dir(nav_beacon, push_destination)
+	travel_distance = get_dist(nav_beacon, push_destination)
+	travel_trip_length = travel_distance
+	idle_platform = push_destination
+	// Don't bother processing crossing signals, where this tram's going there are no signals
+	for(var/obj/machinery/crossing_signal/xing as anything in GLOB.tram_signals)
+		xing.temp_malfunction()
+	priority_announce("In a turn of rather peculiar events, it appears that a high-velocity flying rod (don't ask us where it came from) has led to a station brakes failure on one of the platforms.\n\n\
```

Comment:
The rod is neither flying nor does it have a high velocity, it's the space station that is moving while the rod is completely stationary 🤔 
I don't really care about this one very much though.

Reply:
Yesss good point.